### PR TITLE
[CU-3153c9f] Moved fNFT collection and share asset IDs into staking

### PR DESCRIPTION
**Pull request by PoisonPhang**
## Issue
[[CU-3153c9f]](https://app.clickup.com/t/3153c9f)

## Description
Had to change `FinancialNftProtocl::value_of` in staking because it incorrectly queried pools by fNFT collection ID rather than stake asset ID.

Added function to `CurrencyFactory` trait `expect_create()`, which creates an asset ID without a transactional layer so it can be used within genesis builds of other pallets.

NOTE: All tests that rely on `with_stake` and use its optional `claim` are incorrect and should be rewritten with correct values. This is outside the scope of this PR.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-3153c9f Move fNFT and xToken ID aquesition into staking rewards](https://app.clickup.com/t/3153c9f)

File: code/parachain/frame/composable-traits/src/fnft.rs [hunk 1]
```diff
@@ -56,6 +56,7 @@ pub trait FinancialNftProtocol {
 	/// - collection: id of the financial NFT collection issued/used by the protocol.
 	/// TODO (vim): Think how to represent the difference between assets and liabilities
 	fn value_of(
+		pool_id: &Self::AssetId,
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/fnft.rs:59]: This must not be required, breaks the interface if third parties who has no idea about the pools to find a pool id

**Review by vimukthi-git — CHANGES_REQUESTED** [review]

**Comment by vimukthi-git** [comment]
> Had to change FinancialNftProtocl::value_of in staking because it incorrectly queried pools by fNFT collection ID rather than stake asset ID.

Yeah definitely a bug though, the current change is incorrect, we need to somehow map fnft collection id to pool and execute the operation

File: code/parachain/frame/composable-traits/src/fnft.rs (hunk 1 above)
Comment (PoisonPhang) [reply code/parachain/frame/composable-traits/src/fnft.rs:59]: I see; I'll make a workaround so the interface can still be neutral to the underlying design

**Comment by PoisonPhang** [comment]
On hold while other tasks are prioritized.

**Comment by PoisonPhang** [comment]
PR no longer relevant

**Pull request closed without merging**